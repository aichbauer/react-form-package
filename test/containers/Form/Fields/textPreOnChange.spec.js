import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with text input | preOnChange | manipulate the input value to be duplicated', () => {
  const fieldSpy = jest.spyOn(Form.prototype, 'handleOnChange');
  const event = {
    target: {
      id: 'text',
      value: 'hi',
    },
  };
  const rules = {
    type: 'text',
  };

  const myComponent = (
    <Form>
      <div>
        <div>text</div>
        <Field type="text" id="text" preOnChange={(value) => `${value} ${value}`} />
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(state.data.text.value).toBe('');

  wrapper.find('input').simulate('focus');
  wrapper.find('input').simulate('change', event, rules);
  wrapper.find('input').simulate('blur');

  state = wrapper.state();

  expect(state.data.text.value).toBe('hi hi');
  expect(fieldSpy).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
