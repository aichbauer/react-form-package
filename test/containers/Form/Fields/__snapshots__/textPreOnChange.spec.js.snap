// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with text input | preOnChange | manipulate the input value to be duplicated 1`] = `
<form
  className="rfp-form"
>
  <div>
    <div>
      text
    </div>
    <input
      className="rfp-input"
      id="text"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;
